util/migrate/allocator.py

```diff
@@ -10,11 +11,18 @@
 # Read the DEBUGLOG environment variable
 debug_log = os.getenv("DEBUGLOG", "false").lower() == "true"
 
-# Set the logging level based on DEBUGLOG
-if debug_log:
-    logging.basicConfig(level=logging.DEBUG)
-else:
-    logging.basicConfig(level=logging.INFO)
+
+# Define the logging format to include worker_name and stream_type
+log_format = "%(asctime)s [%(process)d] [%(levelname)s] [%(name)s] %(message)s"
+
+# Set the logging level and format based on DEBUGLOG
+
+# Set up logging
+logging.basicConfig(
```

Comment:
Should we be using [basicConfig](https://docs.python.org/3/library/logging.html#logging.basicConfig) here? It seems like this configures the root logger which might not be what we want

Reply:
You are right. logging.basicConfig affects all logging calls and we want granular control for module util.migrate.allocator. Doing it with [setLevel](https://docs.python.org/3/library/logging.html#logging.Logger.setLevel) with handlers and formatters